.evergreen.yml

```diff
@@ -241,94 +200,40 @@ functions:
       params:
         working_dir: src
         shell: bash
+        env:
+          <<: *compass-env
+          # TODO: can we move these to compass-env?
+          DEBUG: ${debug}
+          npm_config_loglevel: ${npm_loglevel}
+          COMPASS_DISTRIBUTION: ${compass_distribution}
         script: |
           set -e
-          source ~/compass_env.sh
 
-          cat <<EOF_BUILD_SH > ~/compass_package.sh
-          #! /usr/bin/env bash
-          cd $(pwd)
-
-          # Load environment variables
-          source ~/compass_env.sh
-
-          # For debugging any problems when the notary service fails.
-          # @see https://github.com/mongodb-js/notary-service-client For details on what notary service is.
-          export DEBUG=mongodb-notary*
-
-          # Required to sign release assets
-          export NOTARY_URL="http://notary-service.build.10gen.cc:5000"
-          export NOTARY_AUTH_TOKEN="${signing_auth_token}"
-          export NOTARY_SIGNING_KEY="${signing_key_name}"
-          export NOTARY_SIGNING_COMMENT="Evergreen project mongodb/compass ${revision} - ${build_variant} - ${branch_name}"
+          eval $(.evergreen/print-compass-env.js)
 
           if [ `uname` == Darwin ]; then
-            # We do print the environment during the build process in a few places
-            # but we probably should stop doing that. After that, we can go back
-            # to using plain APPLE_USERNAME and APPLE_PASSWORD environment variables.
-            echo '{"appleUsername": "${apple_username}", "applePassword": "${apple_password}"}' > /tmp/compass-apple-cred.json
-            export APPLE_CREDENTIALS_FILE="/tmp/compass-apple-cred.json"
-          fi
-
-          echo "Creating signed release build..."
-
-          if [[ "$OSTYPE" == "cygwin" ]]; then
-            echo "Starting Installer Service..."
-            net start MSIServer
-          fi
-
-          # Provide a verbose logging for the release process
-          export npm_config_loglevel="${npm_loglevel}"
-          export DEBUG="${debug}"
+            # write env vars (including secrets) to a script so we can restore them after running ssh
+            .evergreen/export-compass-env.js >> env-vars.sh
```

Comment:
Can't really see a better way to restore the environment variables once we ssh for mac. Hopefully we can remove this hack soon once we switch to the new notary service.